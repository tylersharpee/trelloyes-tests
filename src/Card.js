import React from "react";
import store from './store';

function Card(props) {
  return (
    <div className="Card">
      <button type="button">delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  );
}

export default Card;
