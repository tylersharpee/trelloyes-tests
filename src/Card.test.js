import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

//Snapshot Test
it ('renders the ui as expected', () => {
  const tree = renderer
    .create(<Card key='5' title='testCardTitle' content='testCardContent'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//Smoke Test
it ('render without crashing', () => {
  //create a DOM element to render the component
  const div = document.createElement('div');

  //render the component
  ReactDOM.render(<Card key='5' title='testCardTitle' content='testCardContent'/>, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
})