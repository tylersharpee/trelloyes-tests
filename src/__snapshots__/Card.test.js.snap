// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ui as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    testCardTitle
  </h3>
  <p>
    testCardContent
  </p>
</div>
`;
